SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -244,40 +244,53 @@ void USpatialActorChannel::Init(UNetConnection* InConnection, int32 ChannelIndex
 	Receiver = NetDriver->Receiver;
 }
 
-void USpatialActorChannel::DeleteEntityIfAuthoritative()
+void USpatialActorChannel::RetireEntityIfAuthoritative()
 {
 	if (NetDriver->Connection == nullptr)
 	{
 		return;
 	}
 
-	bool bHasAuthority = NetDriver->IsAuthoritativeDestructionAllowed() && NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialGDK::Position::ComponentId);
+	if (!NetDriver->IsAuthoritativeDestructionAllowed())
+	{
+		return;
+	}
 
-	if (bHasAuthority)
+	const bool bHasAuthority = NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialGDK::Position::ComponentId);
+	if (Actor != nullptr)
 	{
-		if (Actor != nullptr)
+		if (bHasAuthority)
 		{
-			// Workaround to delay the delete entity request if tearing off.
-			// Task to improve this: UNR-841
-			if (Actor->GetTearOff())
-			{
-				NetDriver->DelayedRetireEntity(EntityId, 1.0f, Actor->IsNetStartupActor());
-				// Since the entity deletion is delayed, this creates a situation,
-				// when the Actor is torn off, but still replicates.
-				// Disabling replication makes RPC calls impossible for this Actor.
-				Actor->SetReplicates(false);
-			}
-			else
+			if (Actor != nullptr)
 			{
-				Sender->RetireEntity(EntityId, Actor->IsNetStartupActor());
+				// Workaround to delay the delete entity request if tearing off.
+				// Task to improve this: UNR-841
+				if (Actor->GetTearOff())
```

Comment:
This is just inverted around Actor != nullptr and hasn't functionally changed. Additional code is added below if (bCreatedEntity)